samples/cellular/udp/Kconfig

```diff
@@ -14,6 +14,14 @@ config UDP_DATA_UPLOAD_FREQUENCY_SECONDS
 	int "How often data is transmitted to the server"
 	default 900
 
+config UDP_DATA_UPLOAD_ITERATIONS
+	int "Number of data transmissions to the server before shutdown"
+	range -1 2147483647
+	default -1
```

Comment:
I think the sample should terminate by default, so that we can run it in CI.

Reply:
@MechanicV1 From CI point of view should this sample terminate by default?

Reply:
CI can change the Kconfig value when building it. Though many other samples are terminating by default.

Reply:
Test automation will take care of setting the Kconfig ! @MirkoCovizzi

Reply:
I think we should implement the same behavior as other sample. It's easy enough to change that for users if needed.
CI might want to override the value anyway to shorten the time it takes to complete the test.